From 3d8ea96eac8a68f89a67263be1f0d3ddd71a5563 Mon Sep 17 00:00:00 2001
From: "Dr. Stefan Schimanski" <stefan.schimanski@gmail.com>
Date: Thu, 29 Oct 2020 13:56:01 +0100
Subject: [PATCH] UPSTREAM: <carry>: noderestrictions: add
 node-role.kubernetes.io/* to allowed node labels

Server side validation of node labels was added in https://github.com/kubernetes/kubernetes/pull/90307. We only disabled kubelet-side validation before to make our node role labels work.

UPSTREAM: <carry>: add control plane to allow roles

OpenShift-Rebase-Source: 38bfed34a3b
OpenShift-Rebase-Source: aff4434f177
---
 staging/src/k8s.io/kubelet/pkg/apis/well_known_labels.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/staging/src/k8s.io/kubelet/pkg/apis/well_known_labels.go b/staging/src/k8s.io/kubelet/pkg/apis/well_known_labels.go
index 96890b553f7..b77b526392d 100644
--- a/staging/src/k8s.io/kubelet/pkg/apis/well_known_labels.go
+++ b/staging/src/k8s.io/kubelet/pkg/apis/well_known_labels.go
@@ -45,6 +45,12 @@ var kubeletLabels = sets.NewString(
 
 	LabelOS,
 	LabelArch,
+
+	// These are special for OpenShift:
+	"node-role.kubernetes.io/control-plane",
+	"node-role.kubernetes.io/master",
+	"node-role.kubernetes.io/worker",
+	"node-role.kubernetes.io/etcd",
 )
 
 var kubeletLabelNamespaces = sets.NewString(
-- 
2.43.0

